### ISSUE TITLE
Chapter 6 OnBeginOverlap of gold coin crashes Unreal Engine

### ISSUE THREAD

**Issue by dredgeho** [PacktPublishing/Multiplayer-Game-Development-with-Unreal-Engine-5#5]
Issue 1: I run the game in Unreal after adding gold coins and when the player runs into the collision Unreal crashes. In Visual Studio it is not showing any errors on the BasePickup. It does however show errors in files I have never edited.  

Error (active)	E1835	attribute "deprecated" does not apply here	UnrealShadows_LOTL	D:\EpicGames\UE_5.1\Engine\Source\Runtime\CoreUObject\Public\Serialization\BulkData.h	283		

Error (active)	E1835	attribute "deprecated" does not apply here	UnrealShadows_LOTL	D:\EpicGames\UE_5.1\Engine\Source\Runtime\CoreUObject\Public\Serialization\BulkData.h	1239		

Etc... 

I don't know if those errors above are related to the crashing on overlap. I've looked these errors up and people say to just ignore them which I find very odd. 

![errors pickup](https://github.com/user-attachments/assets/9d0dfd06-219c-453e-b3a5-fd60d0b64e66)

![crash on pickup](https://github.com/user-attachments/assets/adf815d3-b3eb-42a4-86db-1865d6bfeeeb)

Issue 2: The Event Pickup and AddXP will show errors and I have to live compile every time I reopen the project. Then I have to refresh the nodes to get them to not have errors.  

![event pickup refrash nodes](https://github.com/user-attachments/assets/35ecf42b-b84c-4079-892b-3f0465f361bc)

Issue 3: The issue I posted a day or two ago is gone. It was about adding a header file in visual studio which gave me 9740 errors (no joke).  The book did not say how to add the header file in. I added it by using a text file .  I'm wondering why that issue was deleted. I don't see it moved anywhere. 

Update and issue 4: I copied the files from:
Chapter 06\unrealshadows-ltol-chapter-06-end\unrealshadows-ltol-chapter-06-end
into my source directory thinking this will fix all the errors. But now build fails in Unreal and I still get 15 errors in file never edited in VSC++.  

![build fail](https://github.com/user-attachments/assets/189904a0-b0e5-455b-a28f-24a0d378ae04)

**Comment by dredgeho** [comment]
I copied my files back over and closed unreal and then compiled inside vsc++ by using 'build UnrealShadows" and it compiled. Went back into Unreal and it worked. It did not crash when colliding.

**Comment by marcosecchi** [comment]
Hello @dredgeho regarding:

_Issue 3: The issue I posted a day or two ago is gone. It was about adding a header file in visual studio which gave me 9740 errors (no joke). The book did not say how to add the header file in. I added it by using a text file . I'm wondering why that issue was deleted. I don't see it moved anywhere._

You probably posted it on the wrong repository: https://github.com/marcosecchi/unrealshadows-ltol/issues/1
